src/other/index.jl

```diff
@@ -219,6 +219,8 @@ end
 @inline Base.getindex(x::AbstractIndex, idx::Between) = x[idx.first]:x[idx.last]
 @inline Base.getindex(x::AbstractIndex, idx::All) =
     isempty(idx.cols) ? (1:length(x)) : union(getindex.(Ref(x), idx.cols)...)
+@inline Base.getindex(x::AbstractIndex, idx::Cols) =
+    isempty(idx.cols) ? Int[] : union(getindex.(Ref(x), idx.cols)...)
```

Comment:
This might be enough. At least something like `getindex.(Ref(x), Symbol[])` gives `Int[]`.
```suggestion
    union(getindex.(Ref(x), idx.cols)...)
```

Reply:
It is not OK, as:
```
julia> getindex.(Ref(index(df)), DataAPI.Cols().cols)
()
```
(in short `Cols()` stores an empty tuple not `Symbol[]`)

Reply:
and:
```
julia> union(getindex.(Ref(idx), DataAPI.Cols().cols))
Union{}[]
```